src/Vibrato/Vibrato.cpp

```diff
@@ -0,0 +1,194 @@
+#include "Vibrato.h"
+
+CVibrato::CVibrato(float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels): m_isInitialised(false)
+{
+    init(fDelayInSec, fWidthInSec, fSampleRateInHz, fFrequencyInHz, numChannels);
+}
+
+CVibrato::~CVibrato()
+{
+    this->reset();
+}
+
+Error_t CVibrato::create (CVibrato*& pCInstance, float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels )
+{
+    pCInstance = new CVibrato(fDelayInSec, fWidthInSec, fSampleRateInHz, fFrequencyInHz, numChannels );
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::destroy (CVibrato*& pCInstance)
+{
+    if (!pCInstance)
+        return Error_t::kUnknownError;
+    delete pCInstance;
+    pCInstance = nullptr;
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::init(float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels )
+{
+        m_fSampleRateInHz = fSampleRateInHz;
+        m_iNumChannels = numChannels;
+
+        setParam(kDelay, fDelayInSec);
+        setParam(kWidth, fWidthInSec);
+
+//        this->reset();
```

Comment:
I think you can bring reset() back and remove some of the code below. Should work either way but might as well use the function